### PR TITLE
Calendars clear importer problems on source value change

### LINKED ISSUE #1601: [Bug]: Calendar update status incorrectly persists after URL is changed
## Description

Trying to update the Yard's calendar and it's showing me incorrect older error messages

## Steps to reproduce

1. Replace the URL with a new feed
2. Press save
3. Error shows up - which is the old error
 
This does reset after next import but I think it's very confusing when people are trying to debug this

Screenshot 2022-12-01 at 10 24 46

## What you expected to happen

Errors should reset when the URL field is updated

## A/C
- [ ] When a calendar has an error message on it, changing the source URL and saving will clear that message (and any associated fields)

### PR DESCRIPTION
**Pull request by ivan-kocienski-gfsc**
Closes #1601

### CODE REVIEW

File: app/models/calendar.rb [hunk 1]
```diff
@@ -272,4 +273,12 @@ def check_source_reachable
...
     errors.add :source, 'Unable to autodetect calendar format, please pick an option from the list below'
   end
+
+  def clear_status_on_source_change
+    return unless source_changed?
+
+    self.critical_error = nil
+    self.notices = nil
+    self.last_import_at = nil
+  end
```
Comment (aaaaargZombies) [inline app/models/calendar.rb:283]: This looks like it works though is it missleading to imply the issue has been fixed when it's not been tested yet? @kimadactyl  is this what you wanted?

I suppose the alternative is not clearing the issue and making it clear that you need to wait for the importer to run to findout if the changes have been effective?

happy for this to go in, just double checking before we commit to it.

Comment (kimadactyl) [reply app/models/calendar.rb:283]: yeah hmmm lots of ways you could argue this

i think it prob makes most sense to change the messages to something like "Source URL changed, waiting for queue" and not to blank the last_import?

Comment (ivan-kocienski-gfsc) [reply app/models/calendar.rb:283]: If you feel this is the desired behaviour then the A/C need to be (possibly discussed) and updated

**Review by aaaaargZombies — APPROVED** [review]
Approving based on the AC in the ticket - will discuss at dev standup if it's good AC then we can merge straightaway if everyone is on the same page.

Apologies for throwing doubt on the AC after the work has been done, I appreciate that's frustrating.

**Comment by kimadactyl** [comment]
> Apologies for throwing doubt on the AC after the work has been done, I appreciate that's frustrating.

I think the AC were written after the code was done.

It's fine as is though.

**Comment by aaaaargZombies** [comment]
After discussion Ivan pointed out we have this ticket that is about handling failing imports so the concerns can be covered there #2305

**Pull request merged**